### PR TITLE
Set MySQL rowtype to dynamic for new tables

### LINKED ISSUE #9347: ORM engine initialization Error: migrate: Row size too large
- Gitea version (or commit ref): 1.10.0
- Git version: 2.24.1
- Operating system: Linux (Arch)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:
https://gist.github.com/Whisprin/69ec124abec9fb94fbe713e32f5a652a

## Description

After upgrading from Gitea v1.9.5 to 1.10.0 I'm getting a database migration error:
`ORM engine initialization attempt #8/10 failed. Error: migrate: do migrate: Error 1118: Row size too large. The maximum row size for the used table type, not counting BLOBs, is 8126. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs`

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #9347 for new installations and new tables

### CODE REVIEW

File: models/models.go [hunk 1]
```diff
@@ -138,10 +138,13 @@ func getEngine() (*xorm.Engine, error) {
 		return nil, err
 	}
 
-	engine, err := xorm.NewEngine(setting.Database.Type, connStr)
+	engine, err := xorm.NewEngineWithParams(setting.Database.Type, connStr)
```
Comment (lunny) [inline models/models.go]: `xorm.NewEngineWithParams` needs third params.

Comment (lafriks) [reply models/models.go]: Damn, did not undo enough changes :) This change was not needed

**Review by lunny — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
@lafriks would it be worth just writing a migration so we never have to hear about this or get blamed for this again?

**Comment by guillep2k** [comment]
There's a chance this setting is also required?:

```
SET GLOBAL innodb_file_format=barracuda
```

See [SO](https://stackoverflow.com/a/53605299/225979).

It may depend on the engine version.

**Comment by lunny** [comment]
@zeripath We don't need a migration. Once user encountered this problem, it will install gitea failed.

**Comment by lafriks** [comment]
@guillep2k this needs to be implemented first as param in xorm like it is with rowtype

**Pull request merged**